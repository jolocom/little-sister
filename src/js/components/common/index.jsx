export { default as IconIdCard } from './icon-idcard'
export { default as IconPassport } from './icon-passport'
export { default as IconServices } from './icon-services'
export { default as Loading } from './loading'
export { default as Error } from './error'
export { default as Spinner } from './spinner'
export { default as IconCheckmark } from './icon-checkmark'
export { default as IconEther } from './icon-ether'
export { default as IconBlockchain } from './icon-blockchain'
