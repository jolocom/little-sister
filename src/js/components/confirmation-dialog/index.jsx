import React from 'react'
import { connect } from 'redux/utils'

import { FlatButton, Dialog } from 'material-ui'

@connect({
  props: ['confirm'],
  actions: ['confirmation-dialog:close']
})
export default class ConfirmationDialog extends React.Component {
  static propTypes = {
    close: React.PropTypes.func,
    confirm: React.PropTypes.object
  }

  _handleConfirmAction() {
    this.props.close()
    this.props.confirm.callback() // Action when the user confirms
  }

  _handleConfirmCancel() {
    this.props.close()
  }

  render() {
    const confirmHandler = () => this._handleConfirmAction()
    const cancelHandler = () => this._handleConfirmCancel()

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={cancelHandler}
      />,
      <FlatButton
        label={this.props.confirm.primaryActionText}
        primary
        onTouchTap={confirmHandler}
      />
    ]

    return <Dialog
      actions={confirmActions}
      modal={false}
      open={this.props.confirm.open}
      onRequestClose={this.handleClose}
    >
      {this.props.confirm.message}
    </Dialog>
  }
}
