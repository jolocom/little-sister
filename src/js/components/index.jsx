import React from 'react'
import Radium from 'radium'
import {RaisedButton} from 'material-ui'
import Carousel from 're-carousel'
import IndicatorDots from 'components/common/indicator-dots.jsx'

let Index = React.createClass({
  contextTypes: {
    router: React.PropTypes.object,
    username: React.PropTypes.string,
    muiTheme: React.PropTypes.object
  },

  componentWillMount() {
    const {account} = this.context
    if (account && account.webId) {
      this.context.router.push('/graph')
    }
  },

  getStyles() {
    let {muiTheme} = this.context

    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        backgroundColor: muiTheme.jolocom.gray4
      },
      logo: {
        fontSize: '18px',
        fontWeight: '400',
        textAlign: 'center',
        marginTop: '24px',
        textTransform: 'uppercase'
      },
      logoImg: {
        width: '32px',
        height: '32px',
        verticalAlign: 'middle'
      },
      intro: {
        flex: 1,
        textAlign: 'center'
      },
      slide: {
        backgroundColor: muiTheme.jolocom.gray4,
        padding: '24px',
        height: '100%',
        display: 'flex',
        flexDirection: 'column'
      },
      content: {
        flex: 1
      },
      img: {
        maxWidth: '100%'
      },
      title: {
        color: muiTheme.jolocom.gray1,
        fontSize: '28px',
        fontWeight: '300'
      },
      logoStartImg: {
        padding: '24px',
        maxWidth: '80%',
        width: '320px',
        margin: 'auto',
        marginTop: '20vh'
      },
      actions: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'stretch',
        padding: '24px',
        maxWidth: '80%',
        width: '260px',
        margin: 'auto'
      },
      signup: {
        marginBottom: '12px'
      },
      login: {
      }
    }

    return styles
  },

  render() {
    let styles = this.getStyles()

    return (
      <div style={styles.container}>

        {/** <div style={styles.logo}>
          <img src="/img/logo.png" style={styles.logoImg} /> Jolocom
        </div> **/}

        {/** <Carousel style={styles.intro} indicator={IndicatorDots}>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-data.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Create your personal digital identity</h3>
            <div style={styles.content}>
              <img src="/img/slide-identity.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-share.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Visualize your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-visualize.png" style={styles.img} />
            </div>
          </div>
        </Carousel> **/}

        <div style={styles.intro}>
          <img
            src="/img/logo_littlesister_start.svg"
            style={styles.logoStartImg} />
        </div>

        <div style={styles.actions}>
          <RaisedButton
            secondary
            label="Sign up"
            style={styles.signup}
            onClick={this._handleSignup}
          />
          <RaisedButton
            label="Log in"
            style={styles.login}
            onClick={this._handleLogin}
          />
        </div>
      </div>
    )
  },

  _handleSignup() {
    this.context.router.push('/signup')
  },

  _handleLogin() {
    this.context.router.push('/login')
  }

})

export default Radium(Index)
