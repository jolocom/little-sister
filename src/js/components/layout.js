import Layout from './layout/layout.jsx'
import Content from './layout/content.jsx'
import Spacer from './layout/spacer.jsx'

export {Layout, Content, Spacer}
