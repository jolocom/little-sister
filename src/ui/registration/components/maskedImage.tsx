import React from 'react'
import {
  PanResponder,
  PanResponderInstance,
  GestureResponderEvent,
} from 'react-native'
import { Svg, Path } from 'react-native-svg'
import { JolocomTheme } from 'src/styles/jolocom-theme'

interface Props {
  addPoint: (x: number, y: number) => void
}

interface State {
  currentPath: string[]
  limit: number
}

interface Point {
  type: string
  x: number
  y: number
}

export class MaskedImageComponent extends React.Component<Props, State> {
  private panResponder!: PanResponderInstance

  state = {
    currentPath: [],
    limit: 15,
  }

  componentWillMount() {
    this.panResponder = this.getConfiguredPanResponder()
  }

  private getConfiguredPanResponder(): PanResponderInstance {
    return PanResponder.create({
      onMoveShouldSetPanResponder: () => true,
      onMoveShouldSetPanResponderCapture: () => true,
      onPanResponderGrant: this.handleDrawStart,
      onPanResponderMove: this.handleDraw,
    })
  }

  private handleDrawStart = (e: GestureResponderEvent): void => {
    const { locationX, locationY } = e.nativeEvent
    this.props.addPoint(locationX, locationY)

    const point = { type: 'M', x: locationX, y: locationY }
    this.handleNewPoint(point)
  }

  private handleDraw = (e: GestureResponderEvent): void => {
    const { locationX, locationY } = e.nativeEvent
    this.props.addPoint(locationX, locationY)

    const point = { type: 'L', x: locationX, y: locationY }
    this.handleNewPoint(point)
  }

  private handleNewPoint(p: Point): void {
    const svgCoordinate = `${p.type}${p.x} ${p.y}`
    const newSvgPathCoords: string[] = this.state.currentPath.concat()

    if (newSvgPathCoords.length === this.state.limit) {
      newSvgPathCoords.shift()
      newSvgPathCoords[0] = `M${newSvgPathCoords[0].substring(1)}`
    }

    newSvgPathCoords.push(svgCoordinate)
    this.setState({ currentPath: newSvgPathCoords })
  }

  render() {
    return (
      <Svg width="100%" height="100%" {...this.panResponder.panHandlers}>
        <Path
          d={this.state.currentPath.join(' ')}
          fill="none"
          stroke={JolocomTheme.primaryColorSand}
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={20}
        />
      </Svg>
    )
  }
}
