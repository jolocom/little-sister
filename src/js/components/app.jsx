import React from 'react'
import Reflux from 'reflux'
import Radium from 'radium'
import {bankUri} from 'lib/fixtures'

import {Layout, Content} from 'components/layout'
import {
  Paper,
  AppBar,
  IconButton,
  Snackbar,
  FlatButton,
  Dialog} from 'material-ui'
import Badge from 'material-ui/Badge'
import NavigationMenu from 'material-ui/svg-icons/navigation/menu'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

import JolocomTheme from 'styles/jolocom-theme'

import LeftNav from 'components/left-nav/nav.jsx'
import Profile from 'components/accounts/profile.jsx'
import Tour from 'components/tour.jsx'

import Loading from 'components/common/loading.jsx'

import AccountActions from 'actions/account'
import AccountStore from 'stores/account'
import ConfirmStore from 'stores/confirm'

import PinnedActions from 'actions/pinned'
import ConfirmActions from 'actions/confirm'
import ProfileActions from 'actions/profile'
import ProfileStore from 'stores/profile'

import SnackbarStore from 'stores/snackbar'

// A pathname is considered public if either "/" or if it starts
// with any of the following publicRoutes
const publicRoutes = [
  '/login',
  '/signup',
  '/forgot-password',
  '/change-password'
]

let App = React.createClass({

  mixins: [
    Reflux.connect(AccountStore, 'account'),
    Reflux.connect(ProfileStore, 'profile'),
    Reflux.connect(SnackbarStore, 'snackbar'),
    Reflux.connect(ConfirmStore, 'confirm')
  ],

  propTypes: {
    location: React.PropTypes.object,
    children: React.PropTypes.node,
    route: React.PropTypes.object
  },

  contextTypes: {
    router: React.PropTypes.object.isRequired
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object,
    profile: React.PropTypes.any,
    account: React.PropTypes.object,
    username: React.PropTypes.string,
    searchActive: React.PropTypes.bool,
    location: React.PropTypes.object,
    route: React.PropTypes.object
  },

  getChildContext: function () {
    let {account, profile, searchActive} = this.state
    return {
      muiTheme: this.theme,
      profile,
      account,
      username: account && account.username, // backward compat
      searchActive,
      location: this.props.location,
      route: this.props.route
    }
  },

  getInitialState() {
    return {
      searchActive: false,
      searchQuery: null,
      showIndicator: false
    }
  },

  componentWillMount() {
    this.theme = getMuiTheme(JolocomTheme)
    AccountActions.login()
  },

  componentDidUpdate(prevProps, prevState) {
    let {username} = this.state.account

    if (prevState.account.username === undefined ||
      prevState.account.username !== username) {
      this.checkLogin()
    }
  },

  isPublicRoute(path = this.props.location.pathname) {
    return path === '/' ||
      publicRoutes.some((publicRoute) => path.indexOf(publicRoute) === 0)
  },

  checkLogin() {
    let {username, loggingIn} = this.state.account

    // session is still loading, so return for now
    if (username === undefined && loggingIn) {
      return
    }

    if (!username && !this.isPublicRoute()) {
      this.context.router.push('/login')
    } else if (username && this.isPublicRoute()) {
      this.context.router.push('/graph')
    }

    if (username) {
      ProfileActions.load()
    }
  },

  _handlePinnedTap() {
    PinnedActions.show()
  },

  _handleSearchTap() {
    this.refs.search.show()
    this.setState({searchActive: true})
  },

  _handleSearchChange(query) {
    this.setState({searchQuery: query})
  },

  _handleSearchSubmit() {
    let uri = `${bankUri}/${this.state.searchQuery}#this`
    this.context.router.push(`/graph/${encodeURIComponent(uri)}`)
    this.refs.search.hide()
  },

  _handleSearchHide() {
    this.setState({
      searchActive: false,
      searchQuery: null
    })
  },

  _handleChatTap() {
    this.context.router.push('/conversations')
  },

  showDrawer() {
    this.refs.leftNav.show()
  },

  _handleConfirmClose() {
    ConfirmActions.close()
    this.state.confirm.callback()
  },

  getStyles() {
    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative'
      },
      header: {
        zIndex: 5,
        backgroundColor: this.theme.appBar.color
      },
      bar: {
        boxShadow: 'none'
      },
      nav: {
        display: this.state.searchActive ? 'none' : 'block'
      },
      icon: {
        color: this.theme.appBar.textColor
      },
      filters: {
        width: '100%',
        height: '48px'
      },
      menuIcon: {
        marginTop: '-20px',
        position: 'relative',
        top: '-4px',
        cursor: 'pointer'
      },
      hamburgerBadge: {
        top: 10,
        right: 20,
        width: 15,
        height: 15,
        display: 'none'
      }
    }
    return styles
  },

  render() {
    const styles = this.getStyles()

    // @TODO render login screen when logging in, also makes sures child
    // components don't get rendered before any user data is available
    if (this.state.account.loggingIn && !this.isPublicRoute()) {
      return <Loading />
    }

    // Deactivating search until we get it working
    /* <IconButton
       iconClassName="material-icons"
       iconStyle={styles.icon}
       onTouchTap={this._handleSearchTap}>search</IconButton> */
    const nav = (
      <div>
        <IconButton
          iconClassName="material-icons"
          iconStyle={styles.icon}
          onTouchTap={this._handleChatTap}>chat</IconButton>
      </div>
    )

    // Deactivating search until we get it working
    /*
    (
      <GraphSearch
        ref="search"
        onChange={this._handleSearchChange}
        onSubmit={this._handleSearchSubmit}
        onHide={this._handleSearchHide} />
    )
    */
    const search = null

    // Deactivating the filters until we get them working
    // <GraphFilters style={styles.filters} showDefaults />
    const filters = null

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={this._handleConfirmClose}
      />,
      <FlatButton
        label={this.state.confirm.primaryActionText}
        primary
        onTouchTap={this._handleConfirmClose}
      />
    ]

    return (
      <div style={styles.container}>
        {this.isPublicRoute() ? this.props.children : (
          <Layout>
            <Paper style={styles.header}>
              <AppBar
                title="Graph"
                iconElementRight={nav}
                style={styles.bar}
                iconElementLeft={
                  <Badge
                    badgeContent={''}
                    secondary
                    badgeStyle={styles.hamburgerBadge}>
                    <NavigationMenu
                      onTouchTap={this.showDrawer}
                      style={styles.menuIcon} />
                  </Badge>
                }
                onLeftIconButtonTouchTap={this.showDrawer} />
              {filters}
              {search}
            </Paper>
            <LeftNav ref="leftNav" />
            <Content>
              {React.Children.map(this.props.children, (el) => {
                return React.cloneElement(el, {
                  searchQuery: this.state.searchQuery
                })
              })}
            </Content>
            <Profile />
            <Tour />
          </Layout>
        )}

        <Snackbar
          open={this.state.snackbar.open}
          message={this.state.snackbar.message}
          action={this.state.snackbar.undo && 'undo'}
          onActionTouchTap={this.state.snackbar.undoCallback}
        />

        <Dialog
          actions={confirmActions}
          modal={false}
          open={this.state.confirm.open}
          onRequestClose={this.handleClose}
        >
          {this.state.confirm.message}
        </Dialog>
      </div>

    )
  }

})

export default Radium(App)
