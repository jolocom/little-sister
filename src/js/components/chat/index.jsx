import React from 'react'
import Radium from 'radium'

import {AppBar, Tabs, Tab, Paper, IconButton} from 'material-ui'

import Dialog from 'components/common/dialog.jsx'
import {Layout, Content} from 'components/layout'

import GraphStore from 'stores/graph-store'

@Radium
export default class Chat extends React.Component {
  static contextTypes = {
    router: React.PropTypes.any,
    store: React.PropTypes.object
  }

  static propTypes = {
    children: React.PropTypes.node,
    location: React.PropTypes.object
  }

  constructor(props) {
    super(props)

    this.state = {
      activeTab: this.getActiveTab(props.location.pathname)
    }
  }

  getActiveTab(path) {
    let activeTab = 'chat'

    if (path === '/contacts') {
      activeTab = 'contacts'
    }

    return activeTab
  }

  componentWillUpdate(newProps) {
    if (this.props.location.pathname !== newProps.location.pathname) {
      this.setState({
        activeTab: this.getActiveTab(newProps.location.pathname)
      })
    }
  }

  close() {
    if (GraphStore.state.center == null) {
      this.context.router.push('/graph')
    } else {
      this.context.router.push('/graph/' +
        encodeURIComponent(GraphStore.state.center.uri))
    }
  }

  render() {
    const backIcon = (
      <IconButton
        iconClassName="material-icons"
        iconStyle={styles.icon}
        onTouchTap={this._handleBackTouchTap}
      >
        arrow_back
      </IconButton>
    )

    /*
    const searchIcon = (
      <IconButton
        iconClassName="material-icons"
        iconStyle={styles.icon}
      >
        search
      </IconButton>
    )
    */

    return (
      <Dialog id="chat" visible fullscreen>
        <Layout>
          <Paper>
            <AppBar
              title="Chat"
              style={styles.bar}
              iconElementLeft={backIcon}
              // iconElementRight={searchIcon}
            />
            <Tabs valueLink={{
              value: this.state.activeTab,
              requestChange: (tab) => this._handleTabsChange(tab)
            }}>
              <Tab label="Conversations" value="chat" />
              <Tab label="Contacts" value="contacts" />
            </Tabs>
          </Paper>
          <Content style={styles.conversationsContainer}>
            {this.props.children}
          </Content>
        </Layout>
      </Dialog>
    )
  }

  _handleTabsChange(tab) {
    this.setState({activeTab: tab})

    switch (tab) {
      case 'chat':
        this.context.router.push('/conversations')
        break
      case 'contacts':
        this.context.router.push('/contacts')
        break
    }
  }

  _handleBackTouchTap = () => {
    this.close()
  }
}

let styles = {
  bar: {
    boxShadow: 'none'
  },
  conversationsContainer: {
    overflowY: 'auto'
  }
}
