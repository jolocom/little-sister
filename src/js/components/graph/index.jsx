import React from 'react'
import Radium from 'radium'
import { connect } from 'redux/utils'
import {
  AppBar,
  Paper,
  IconButton,
  Badge
} from 'material-ui'

import {Layout, Content} from 'components/layout'
import LeftNavToggle from 'components/left-nav/toggle'

import UnreadMessagesActions from 'actions/unread-messages'
import UnreadMessagesStore from 'stores/unread-messages'

import {bankUri} from 'lib/fixtures'

import Graph from './graph'

import theme from 'styles/jolocom-theme'

@connect({
  props: ['account']
})
@Radium
export default class GraphScreen extends React.Component {
  static contextTypes = {
    router: React.PropTypes.any,
    store: React.PropTypes.object
  }

  static propTypes = {
    children: React.PropTypes.node,
    location: React.PropTypes.object,
    account: React.PropTypes.object,
    showLeftNav: React.PropTypes.func
  }

  constructor(props) {
    super(props)

    // temp until we move to redux
    this.unreadListener = UnreadMessagesStore.listen(
      this.onUnReadMessageUpdate, this
    )

    this.state = {
      searchActive: false,
      searchQuery: null
    }
  }

  componentWillMount() {
    let {webId} = this.props.account
    if (webId) {
      UnreadMessagesActions.load(webId, true)
    }
  }

  componentWillUnmount() {
    const webId = this.props.account
    if (webId) {
      UnreadMessagesActions.unsubscribe(webId)
    }

    this.unreadListener()
  }

  onUnReadMessageUpdate(unreadMessages) {
    this.setState({unreadMessages})
  }

  getUnreadCount() {
    const {unreadMessages} = this.state
    return unreadMessages && unreadMessages.items &&
      unreadMessages.items.length || 0
  }

  getStyles() {
    return {
      header: {
        zIndex: 5,
        backgroundColor: theme.appBar.color
      },
      bar: {
        boxShadow: 'none'
      },
      nav: {
        display: this.state.searchActive ? 'none' : 'block'
      },
      icon: {
        color: theme.appBar.textColor
      },
      filters: {
        width: '100%',
        height: '48px'
      },
      chatBadge: {
        display: this.getUnreadCount() ? 'flex' : 'none'
      },
      navBadge: {
        padding: 0
      }
    }
  }

  render() {
    const styles = this.getStyles()

    // Deactivating search until we get it working
    /*
    <IconButton
      iconClassName="material-icons"
      iconStyle={styles.icon}
      onTouchTap={this._handleSearchTap}>search</IconButton>
    */
    const nav = (
      <div>
        <Badge
          badgeContent={this.getUnreadCount()}
          secondary
          style={styles.navBadge}
          badgeStyle={styles.chatBadge}
        >
          <IconButton
            iconClassName="material-icons"
            iconStyle={styles.icon}
            onTouchTap={this._handleChatTap}>chat</IconButton>
        </Badge>
      </div>
    )

    // Deactivating search until we get it working
    /*
    (
      <GraphSearch
        ref="search"
        onChange={this._handleSearchChange}
        onSubmit={this._handleSearchSubmit}
        onHide={this._handleSearchHide} />
    )
    */
    // const search = null

    // Deactivating the filters until we get them working
    // <GraphFilters style={styles.filters} showDefaults />
    // const filters = null

    return (
      <Layout>
        <Paper style={styles.header}>
          <AppBar
            title="Graph"
            style={styles.bar}
            iconElementRight={nav}
            iconElementLeft={<LeftNavToggle />}
          />
        </Paper>
        <Content>
          <Graph>
            {this.props.children}
          </Graph>
        </Content>
      </Layout>
    )
  }

  _handleSearchTap = () => {
    this.refs.search.show()
    this.setState({searchActive: true})
  }

  _handleSearchChange = (query) => {
    this.setState({searchQuery: query})
  }

  _handleSearchSubmit = () => {
    let uri = `${bankUri}/${this.state.searchQuery}#this`
    this.context.router.push(`/graph/${encodeURIComponent(uri)}`)
    this.refs.search.hide()
  }

  _handleSearchHide = () => {
    this.setState({
      searchActive: false,
      searchQuery: null
    })
  }

  _handleChatTap = () => {
    this.context.router.push('/conversations')
  }

}
