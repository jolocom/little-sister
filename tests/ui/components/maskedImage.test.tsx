import React from 'react'
import { mount, shallow } from 'enzyme'
import { MaskedImageComponent } from 'src/ui/registration/components/maskedImage'

describe('MaskedImage component', ()=> {
  const props = {
    addPoint: (x: number, y: number) => null 
  }

  it('correctly calls the function to configure PanResponder on component mount', () => {
      const configureSpy = jest.spyOn(MaskedImageComponent.prototype, 'getConfiguredPanResponder')
      const rendered = shallow(<MaskedImageComponent {...props}/>)
      expect(configureSpy).toHaveBeenCalledTimes(1)
  })

  it('matches the snapshot with empty current path string', () => {
    const rendered = shallow(<MaskedImageComponent {...props}/>)
    expect(rendered).toMatchSnapshot()
    expect(rendered.state().currentPath).toEqual([])
    expect(rendered.state().limit).toEqual(15)
  })

  it('correctly handles a gesture start', () => {
    const mockNativeEvent = {
      nativeEvent: {
        locationX: 191.5,
        locationY: 381
      }
    }
    const rendered = shallow(<MaskedImageComponent {...props}/>)
    const instance = rendered.instance()

    instance.handleNewPoint = jest.fn()
    instance.handleDrawStart(mockNativeEvent)

    expect(instance.handleNewPoint).toBeCalledWith({
      type: 'M',
      x: mockNativeEvent.nativeEvent.locationX,
      y: mockNativeEvent.nativeEvent.locationY
    }) 
  })

  it('correctly handles gesture continuation', () => {
    const mockNativeEvent = {
      nativeEvent: {
        locationX: 188,
        locationY: 381
      }
    }

    const rendered = shallow(<MaskedImageComponent {...props}/>)
    const instance = rendered.instance()

    instance.handleNewPoint = jest.fn()
    instance.handleDraw(mockNativeEvent)

    expect(instance.handleNewPoint).toBeCalledWith({
      type: 'L',
      x: mockNativeEvent.nativeEvent.locationX,
      y: mockNativeEvent.nativeEvent.locationY
    }) 
  })

  it('correctly handles adding new points to the current path if the path length is shorter than the limit', () => {
    const rendered = shallow(<MaskedImageComponent {...props}/>)

    rendered.setState({currentPath: ['M191.5 381']})
    rendered.update()
    expect(rendered).toMatchSnapshot()

    rendered.instance().handleNewPoint({ type: 'L', x: 188, y: 381 })
    rendered.update()
    expect(rendered).toMatchSnapshot()
  })

  it('correctly handles adding new points to the current path if the path length is longer than the limit', () => {
    const point = {type: "L", x: 187.5, y: 188.5}
    const rendered = shallow(<MaskedImageComponent {...props}/>)

    const lengthLimit = rendered.state().limit
    const mockMaxLengthPath = Array(lengthLimit).fill('L173 174')

    rendered.setState({currentPath: mockMaxLengthPath})
    rendered.update()
    expect(rendered).toMatchSnapshot()

    rendered.instance().handleNewPoint(point)
    rendered.update()
    expect(rendered).toMatchSnapshot()
  })
})
