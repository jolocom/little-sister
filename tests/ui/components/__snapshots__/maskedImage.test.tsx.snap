// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskedImage component correctly handles adding new points to the current path if the path length is longer than the limit 1`] = `
<Svg
  height="100%"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  preserveAspectRatio="xMidYMid meet"
  width="100%"
>
  <Path
    d="L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174"
    fill="none"
    stroke="#942f51"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={20}
  />
</Svg>
`;

exports[`MaskedImage component correctly handles adding new points to the current path if the path length is longer than the limit 2`] = `
<Svg
  height="100%"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  preserveAspectRatio="xMidYMid meet"
  width="100%"
>
  <Path
    d="M173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L173 174 L187.5 188.5"
    fill="none"
    stroke="#942f51"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={20}
  />
</Svg>
`;

exports[`MaskedImage component correctly handles adding new points to the current path if the path length is shorter than the limit 1`] = `
<Svg
  height="100%"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  preserveAspectRatio="xMidYMid meet"
  width="100%"
>
  <Path
    d="M191.5 381"
    fill="none"
    stroke="#942f51"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={20}
  />
</Svg>
`;

exports[`MaskedImage component correctly handles adding new points to the current path if the path length is shorter than the limit 2`] = `
<Svg
  height="100%"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  preserveAspectRatio="xMidYMid meet"
  width="100%"
>
  <Path
    d="M191.5 381 L188 381"
    fill="none"
    stroke="#942f51"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={20}
  />
</Svg>
`;

exports[`MaskedImage component matches the snapshot with empty current path string 1`] = `
<Svg
  height="100%"
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  preserveAspectRatio="xMidYMid meet"
  width="100%"
>
  <Path
    d=""
    fill="none"
    stroke="#942f51"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={20}
  />
</Svg>
`;
